twisted/python/compat.py

```diff
@@ -765,6 +765,29 @@ def _coercedUnicode(s):
 
 
 
+def _bytesRepr(bytestring):
+    """
+    Provide a repr for a byte string that begins with 'b' on both
+    Python 2 and 3.
+
+    @param bytestring: The string to repr.
+    @type bytestring: L{bytes}
+
+    @raise TypeError: The input is not L{bytes}.
+
+    @return: The repr with a leading 'b'.
+    @rtype: L{bytes}
+    """
+    if not isinstance(bytestring, bytes):
+        raise TypeError("Expected bytes not %r" % (bytestring,))
```

Comment:
I prefer something like this too, though I usually use `repr`/reprlib` for it.  Should `_bytesRepr` do truncation?  If so, I'd prefer it be controlled by an argument that defaults to `False`, and that it use a `repr[lib].Repr` instance that's also retrieved from an argument, e.g.:

```def _bytesRepr(bytestring, truncate=False, reprInstance=Repr()):```

Does that seem OK?